New preferences page layout

roryabraham opened the pull request:
HOLD on https://github.com/Expensify/App/pull/20663

### Details
This PR was split off of https://github.com/Expensify/App/pull/15462. It implements a new layout in the `Settings` -> `Preferences` page.

### Fixed Issues
$ (partial) https://github.com/Expensify/App/issues/12261

### Tests
1. Open the preferences page.
1. Make sure it matches the screenshots.
1. Test the UI for regressions.

- [x] Verify that no errors appear in the JS console

### Offline tests
Same as tests.

### QA Steps
Same as tests.

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/47436092/770c5beb-6904-4624-bf17-99340ca79f31

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://github.com/Expensify/App/assets/47436092/7afe5e17-9966-4e7f-9df2-1b10e7ce4f79

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/47436092/83b059d7-15e0-401b-896c-d74d6294b2ec

</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/47436092/80ffaebe-37db-4f1f-9ac4-d6d9c686ebbc

</details>

<details>
<summary>iOS</summary>

https://github.com/Expensify/App/assets/47436092/35663ad2-cdb1-484d-bee9-76273fc59d3b

</details>

<details>
<summary>Android</summary>

https://github.com/Expensify/App/assets/47436092/5aa687d2-935a-4df3-85d1-e99edcaac38c

</details>

roryabraham committed:
Add asset for PreferencesDJ.json

roryabraham committed:
Aggregate LottieAnimations in a single file

roryabraham committed:
Extract HeaderWithBackButton propTypes to separate file

roryabraham committed:
Convert to functional component

roryabraham committed:
Implement useDelayToggleButtonState hook

roryabraham committed:
Create IllustratedHeaderPageLayout component

roryabraham committed:
Implement IllustratedHeaderPageLayout for the PreferencesPage

roryabraham committed:
Use useLocalize hook

roryabraham committed:
Create useEnvironment hook

roryabraham commented:
This PR is branched off of https://github.com/Expensify/App/pull/20663, so the diff will be smaller once that's merged.

roryabraham committed:
Merge branch 'main' into Rory-PreferencesNewLayout

roryabraham committed:
Use canvas renderer instead of svg

roryabraham commented:
@allroundexperts is going to review this

roryabraham commented on patches/react-native-web-lottie+1.4.4.patch:
```diff
@@ -0,0 +1,21 @@
+       container: this.animationDOMNode,
+       animationData: props.source,
+-      renderer: 'svg',
++      renderer: 'canvas',
```
I found this seems to yield better framerates and also fixes a display but on iOS Safari with different animation (not yet included in the app)

roryabraham commented on src/components/HeaderWithBackButton/index.js:
```diff
@@ -0,0 +1,132 @@
+import React from 'react';
```
IDK why this shows as deleted + new file instead of renamed 😏

OSBotify commented:
Use the links below to test this build in android and iOS. Happy testing!

OSBotify commented:
:test_tube::test_tube: Use the links below to test this build in android and iOS. Happy testing! :test_tube::test_tube:
| android :robot:  | iOS :apple: |
| ------------- | ------------- |
| https://ad-hoc-expensify-cash.s3.amazonaws.com/android/21852/index.html  | https://ad-hoc-expensify-cash.s3.amazonaws.com/ios/21852/index.html  |
| ![Android](https://api.qrserver.com/v1/create-qr-code/?size=120x120&data=https://ad-hoc-expensify-cash.s3.amazonaws.com/android/21852/index.html)  | ![iOS](https://api.qrserver.com/v1/create-qr-code/?size=120x120&data=https://ad-hoc-expensify-cash.s3.amazonaws.com/ios/21852/index.html)  |
| desktop :computer: | web :spider_web: |
| https://ad-hoc-expensify-cash.s3.amazonaws.com/desktop/21852/NewExpensify.dmg  | https://21852.pr-testing.expensify.com  |
| ![Desktop](https://api.qrserver.com/v1/create-qr-code/?size=120x120&data=https://ad-hoc-expensify-cash.s3.amazonaws.com/desktop/21852/NewExpensify.dmg)  | ![Web](https://api.qrserver.com/v1/create-qr-code/?size=120x120&data=https://21852.pr-testing.expensify.com)  |

roryabraham commented:
@allroundexperts refresh the page maybe? GitHub was having issues but I posted all the screenshots before putting this in for review. I can still see them in the issue description

roryabraham commented:
Here's the mockup that @shawnborton provided: 

![image](https://github.com/Expensify/App/assets/47436092/0b38efe4-8f6a-42ab-9fc4-92b2324d86a3)

roryabraham commented:
Good catch @allroundexperts – investigating now

shawnborton commented:
Otherwise the screenshots look pretty good to me!

roryabraham committed:
Merge branch 'main' into Rory-PreferencesNewLayout

# Conflicts:
#	src/components/HeaderWithBackButton.js

roryabraham committed:
use CONST for accessibilityRoleg

roryabraham committed:
Use prop destructuring for IllustratedHeaderPageLayout

roryabraham committed:
Use prop destructuring for HeaderWithBackButton

roryabraham committed:
Remove unused defaultProps

roryabraham committed:
alphabetize props

roryabraham committed:
Wrap header with ScrollView

roryabraham committed:
Fix overscroll bottom issue

roryabraham committed:
Create useOverscrollMeasurement hook

roryabraham commented:
Just updated the scrolling behavior (among a few other small fixes). I am aware of a minor issue on iOS and iOS Safari where the animation can overscroll beyond its blue header:

https://github.com/Expensify/App/assets/47436092/e4314b6f-a7ae-48b6-b050-ade563f8b244

I fixed the same issue on the bottom (where before you could overscroll on the bottom and see some blue peeking out). However, fixing the same thing on the top caused scroll jankiness b/c the "spacer view" that we need to expand to keep blue beneath the animation must start w/ height 0, and then we can't possibly re-render it in JS fast enough to keep up with native scroll events.

Anyways, I don't think it's really a problem so this should be ready for another review.

roryabraham committed:
Remove unused import

roryabraham committed:
Remove inline style

OSBotify commented:
:test_tube::test_tube: Use the links below to test this build in android and iOS. Happy testing! :test_tube::test_tube:
| android :robot:  | iOS :apple: |
| ------------- | ------------- |
| https://ad-hoc-expensify-cash.s3.amazonaws.com/android/21852/index.html  | https://ad-hoc-expensify-cash.s3.amazonaws.com/ios/21852/index.html  |
| ![Android](https://api.qrserver.com/v1/create-qr-code/?size=120x120&data=https://ad-hoc-expensify-cash.s3.amazonaws.com/android/21852/index.html)  | ![iOS](https://api.qrserver.com/v1/create-qr-code/?size=120x120&data=https://ad-hoc-expensify-cash.s3.amazonaws.com/ios/21852/index.html)  |
| desktop :computer: | web :spider_web: |
| https://ad-hoc-expensify-cash.s3.amazonaws.com/desktop/21852/NewExpensify.dmg  | https://21852.pr-testing.expensify.com  |
| ![Desktop](https://api.qrserver.com/v1/create-qr-code/?size=120x120&data=https://ad-hoc-expensify-cash.s3.amazonaws.com/desktop/21852/NewExpensify.dmg)  | ![Web](https://api.qrserver.com/v1/create-qr-code/?size=120x120&data=https://21852.pr-testing.expensify.com)  |

shawnborton commented:
Looks like we get some weirdness with the screen height is not very tall:
<img width="385" alt="image" src="https://github.com/Expensify/App/assets/2319350/436b81c6-bdc4-44c1-b836-d5a6e08cce12">


That blue BG part should just remain the same height.

And of course this is an extreme example, but if you zoom out a lot, the blue background is also wonky:
<img width="152" alt="image" src="https://github.com/Expensify/App/assets/2319350/ef64c319-2609-4d4c-b501-af6d0c9b7860">

Maybe we need to take the flex off of it and makes sure it has a fixed height that doesn't shrink or something.

shawnborton commented:
For the icon:
<img width="191" alt="image" src="https://github.com/Expensify/App/assets/2319350/8cb602e7-12e3-4572-8af7-15adf54fffa0">


I think we should consider making that a different color. Either the same color as text in dark mode, or maybe we make it ivory, or ivory with a slightly about of transparency. I think my vote would be to make the icons ivory @ 80% opacity, and the text ("Preferences") in ivory. This way it will play nicely across both themes.

roryabraham commented:
@shawnborton thanks for pointing that out. I'll work more on the layout issues. A couple clarifying questions about icon color:

- Are you referring only to the icon in this header, or the global icon color for icons everywhere?
- A best practice that we've established in many places across the app is that pressable things (such as the back button) should have three different colors depending on their state (default, hovered, pressed). If the default color is ivory w/ 80% opacity, what should be the hover color and the pressed color?

Thanks!

roryabraham commented:
> Just icons in any colorful/styled empty state header

Gotcha, thanks 👍🏼

> Do our current back buttons have a hover or press style?

Looks like you're right. NAB then, but if you do want to play with it and come up with some values you like it would be an easy improvement to throw in here. Otherwise I'll just use ivory w/ opacity 80%

shawnborton commented:
Let's start with ivory @ 80% and adjust from there. We might need to make adjustments after seeing all of the other background colors, too. 

What are the next steps for getting these empty states in other places?

roryabraham committed:
Merge branch 'main' into Rory-PreferencesNewLayout

roryabraham commented:
Well after this is merged then several other pages (security, payments, help, workspaces) should be pretty trivial.

The ones that will be more involved are the ones where the header includes more than just the animation, such as the initial settings page and the About page.

roryabraham committed:
Implement layout adjustments

roryabraham committed:
Hide scroll indicators (for macOS safari)

roryabraham commented:
I think I fixed the layout issues @shawnborton pointed out, but as of now there is a bug here with the wrong color of overscroll on iOS Safari:

https://github.com/Expensify/App/assets/47436092/f96c01cb-2d78-4929-bb97-0f6ea276bd97

roryabraham committed:
Fix overscroll issue on iOS

roryabraham committed:
Fix header color regression

roryabraham committed:
Add ivory for icon and title color

roryabraham commented:
Switched icon and title color to ivory:

<img width="385" alt="image" src="https://github.com/Expensify/App/assets/47436092/251227bf-e13d-46a8-87e5-ab61e9c09f97">

shawnborton commented:
Nice! Can you try it at reduced opacity, like 80%?

OSBotify commented:
:test_tube::test_tube: Use the links below to test this build in android and iOS. Happy testing! :test_tube::test_tube:
| android :robot:  | iOS :apple: |
| ------------- | ------------- |
| https://ad-hoc-expensify-cash.s3.amazonaws.com/android/21852/index.html  | https://ad-hoc-expensify-cash.s3.amazonaws.com/ios/21852/index.html  |
| ![Android](https://api.qrserver.com/v1/create-qr-code/?size=120x120&data=https://ad-hoc-expensify-cash.s3.amazonaws.com/android/21852/index.html)  | ![iOS](https://api.qrserver.com/v1/create-qr-code/?size=120x120&data=https://ad-hoc-expensify-cash.s3.amazonaws.com/ios/21852/index.html)  |
| desktop :computer: | web :spider_web: |
| https://ad-hoc-expensify-cash.s3.amazonaws.com/desktop/21852/NewExpensify.dmg  | https://21852.pr-testing.expensify.com  |
| ![Desktop](https://api.qrserver.com/v1/create-qr-code/?size=120x120&data=https://ad-hoc-expensify-cash.s3.amazonaws.com/desktop/21852/NewExpensify.dmg)  | ![Web](https://api.qrserver.com/v1/create-qr-code/?size=120x120&data=https://21852.pr-testing.expensify.com)  |

shawnborton commented:
Nice, the height issue seems to be fixed.

roryabraham committed:
Use opacity 0.8

roryabraham commented:
Ah, missed that detail. Fixed now:

<img width="378" alt="image" src="https://github.com/Expensify/App/assets/47436092/5ccc3976-43eb-4d8d-89d7-3d76171d71fb">

OSBotify commented:
:test_tube::test_tube: Use the links below to test this build in android and iOS. Happy testing! :test_tube::test_tube:
| android :robot:  | iOS :apple: |
| ------------- | ------------- |
| https://ad-hoc-expensify-cash.s3.amazonaws.com/android/21852/index.html  | https://ad-hoc-expensify-cash.s3.amazonaws.com/ios/21852/index.html  |
| ![Android](https://api.qrserver.com/v1/create-qr-code/?size=120x120&data=https://ad-hoc-expensify-cash.s3.amazonaws.com/android/21852/index.html)  | ![iOS](https://api.qrserver.com/v1/create-qr-code/?size=120x120&data=https://ad-hoc-expensify-cash.s3.amazonaws.com/ios/21852/index.html)  |
| desktop :computer: | web :spider_web: |
| https://ad-hoc-expensify-cash.s3.amazonaws.com/desktop/21852/NewExpensify.dmg  | https://21852.pr-testing.expensify.com  |
| ![Desktop](https://api.qrserver.com/v1/create-qr-code/?size=120x120&data=https://ad-hoc-expensify-cash.s3.amazonaws.com/desktop/21852/NewExpensify.dmg)  | ![Web](https://api.qrserver.com/v1/create-qr-code/?size=120x120&data=https://21852.pr-testing.expensify.com)  |

roryabraham commented:
@allroundexperts I think this is ready for more reviews when you get a chance

roryabraham committed:
Merge branch 'main' into Rory-PreferencesNewLayout

roryabraham commented on src/hooks/useEnvironment.js:
```diff
@@ -0,0 +1,19 @@
+import {useState, useEffect} from 'react';
+import CONST from '../CONST';
+import * as Environment from '../libs/Environment/Environment';
+
+export default function useEnvironment() {
+    const [environment, setEnvironment] = useState(CONST.ENVIRONMENT.PRODUCTION);
+    const [environmentURL, setEnvironmentURL] = useState(CONST.NEW_EXPENSIFY_URL);
+
+    useEffect(() => {
+        Environment.getEnvironment().then(setEnvironment);
+        Environment.getEnvironmentURL().then(setEnvironmentURL);
+    }, []);
+
+    return {
+        environment,
+        environmentURL,
+        isProduction: environment === CONST.ENVIRONMENT.PRODUCTION,
+    };
+}
```
Good idea. I was thinking this would be very quick but on Android it actually makes a call to the GitHub API so yeah we should improve this.

roryabraham replied:
Updated to use context

roryabraham committed:
Create context for environment

roryabraham committed:
Remove shiftUpOnePixel style

roryabraham commented on src/components/IllustratedHeaderPageLayout.js:
```diff
@@ -0,0 +1,68 @@
+                    />
+                    <View style={[styles.flex1, StyleUtils.getBackgroundColorStyle(themeColors.appBG)]}>
+                        <ScrollView
+                            style={[styles.shiftUpOnePixel]}
```
Looks like this was a remnant from another approach I was attempting and isn't needed anymore. Removed.

roryabraham committed:
Add missing useEffect dependency

roryabraham commented on src/hooks/useDelayToggleButtonState.js:
```diff
@@ -0,0 +1,28 @@
+            return;
+        }
+        clearTimeout(resetButtonStateCompleteTimer.current);
+    });
```
FWIW this was just me converting an existing HOC to a hook. Good catch on the missing `useEffect` dependency though.

roryabraham replied:
Ended up scope-creeping a bit to clean this up because I found it confusing. I mostly followed your suggestion

roryabraham committed:
Improve clarity of hook, now called useThrottledButtonState

roryabraham committed:
Remove unused autoReset prop

roryabraham committed:
Update PressableWithDelayToggle to use new hook

roryabraham committed:
Merge branch 'main' into Rory-PreferencesNewLayout

# Conflicts:
#	src/App.js

OSBotify commented:
:test_tube::test_tube: Use the links below to test this build in android and iOS. Happy testing! :test_tube::test_tube:
| android :robot:  | iOS :apple: |
| ------------- | ------------- |
| ❌ FAILED ❌  | ❌ FAILED ❌  |
| The QR code can't be generated, because the android build failed  | The QR code can't be generated, because the iOS build failed  |
| desktop :computer: | web :spider_web: |
| ❌ FAILED ❌  | ❌ FAILED ❌  |
| The QR code can't be generated, because the Desktop build failed  | The QR code can't be generated, because the web build failed  |

roryabraham committed:
Use useContext in withEnvironment

roryabraham committed:
Memoize contextValue

roryabraham committed:
Change propTypes name back to reduce diff

roryabraham committed:
Fix Sidebar tests

roryabraham committed:
Add missing propType comment

roryabraham commented on src/components/Pressable/PressableWithDelayToggle.js:
```diff
@@ -1,22 +1,19 @@
 import variables from '../../styles/variables';
 import getButtonState from '../../libs/getButtonState';
 import * as StyleUtils from '../../styles/StyleUtils';
-import withLocalize, {withLocalizePropTypes} from '../withLocalize';
```
No localization props were used here, so this wasn't needed anymore

roryabraham commented on src/components/withDelayToggleButtonState.js:
```diff
@@ -29,18 +29,11 @@ export default function (WrappedComponent) {
-        /**
-         * @param {Boolean} resetAfterDelay Impose delay before toggling state
-         */
-        toggleDelayButtonState(resetAfterDelay) {
```
`resetAfterDelay` was always, true, so I removed this

roryabraham commented on src/components/withEnvironment.js:
```diff
@@ -12,53 +14,49 @@ const environmentPropTypes = {
+        Environment.getEnvironmentURL().then(setEnvironmentURL);
+    }, []);
+
+    const contextValue = useMemo(
```
Why did I memoize this you ask?

Because if you pass an inline object directly to the context provider like so:

```
<EnvironmentContext.Provider value={{environment, environmentURL}}></EnvironmentContext.Provider>
```

Then the object passed to value `{environment, environmentURL}` is not referentially stable and all the context consumers will be updated on every re-render of the provider, even if the values for `environment` and `environmentURL` don't change.

To fix that, wrap the context value in `useMemo`

roryabraham committed:
Remove unused useOverscrollMeasurement hook

roryabraham commented on src/pages/home/report/ContextMenu/BaseReportActionContextMenu.js:
```diff
@@ -120,7 +120,6 @@ class BaseReportActionContextMenu extends React.Component {
                                 key={contextAction.textTranslateKey}
                                 onPress={() => interceptAnonymousUser(() => contextAction.onPress(closePopup, payload))}
                                 description={contextAction.getDescription(this.props.selection, this.props.isSmallScreenWidth)}
-                                autoReset={contextAction.autoReset}
```
This was always true, so I removed the `autoReset` prop entirely

roryabraham commented:
Thanks for the feedback @allroundexperts, made some improvements + cleanup here. Ready for another round of reviews 🙇🏼

OSBotify commented:
Use the links below to test this build in android and iOS. Happy testing!

OSBotify commented:
:test_tube::test_tube: Use the links below to test this build in android and iOS. Happy testing! :test_tube::test_tube:
| android :robot:  | iOS :apple: |
| ------------- | ------------- |
| https://ad-hoc-expensify-cash.s3.amazonaws.com/android/21852/index.html  | https://ad-hoc-expensify-cash.s3.amazonaws.com/ios/21852/index.html  |
| ![Android](https://api.qrserver.com/v1/create-qr-code/?size=120x120&data=https://ad-hoc-expensify-cash.s3.amazonaws.com/android/21852/index.html)  | ![iOS](https://api.qrserver.com/v1/create-qr-code/?size=120x120&data=https://ad-hoc-expensify-cash.s3.amazonaws.com/ios/21852/index.html)  |
| desktop :computer: | web :spider_web: |
| https://ad-hoc-expensify-cash.s3.amazonaws.com/desktop/21852/NewExpensify.dmg  | https://21852.pr-testing.expensify.com  |
| ![Desktop](https://api.qrserver.com/v1/create-qr-code/?size=120x120&data=https://ad-hoc-expensify-cash.s3.amazonaws.com/desktop/21852/NewExpensify.dmg)  | ![Web](https://api.qrserver.com/v1/create-qr-code/?size=120x120&data=https://21852.pr-testing.expensify.com)  |

roryabraham committed:
Merge branch 'main' into Rory-PreferencesNewLayout

# Conflicts:
#	src/App.js
#	src/components/HeaderWithBackButton.js
#	src/components/withEnvironment.js
#	src/hooks/useEnvironment.js
#	src/pages/home/report/ContextMenu/BaseReportActionContextMenu.js
#	src/pages/settings/Preferences/PreferencesPage.js

roryabraham committed:
Apply changes from HeaderWithBackButton

roryabraham commented:
Resolved conflicts, let's get this over the finish line 😃

roryabraham committed:
Remove outdated localization props from PreferencesPage

roryabraham committed:
Remove unused prop

roryabraham commented:
@allroundexperts all fixed now

allroundexperts reviewed (approved):
Solid work @roryabraham!

I have left some minor comments. However, you can merge this if you feel like those are not urgent enough to be addressed.

MonilBhavsar reviewed (approved):
Looks good to me.

roryabraham commented on src/components/withEnvironment.js:
There are several counter-examples, where the context is created in the HOC, such as in `withLocalize`, `withCurrentReportID`, `withKeyboardState`, `withWindowDimensions`, so I'd say we're not consistent yet.

Perhaps in a follow-up we can standardize on moving all contexts to `src/contexts` and having a file for each 👍🏼

roryabraham commented:
TIL about `git mv`, thanks @MonilBhavsar!

and thanks for the comments @allroundexperts. I think there's a number of good thoughts there but it's more important to get this merged so I can implement the other pages soon here.

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/5579799089) to staging by https://github.com/roryabraham in version: 1.3.42-0 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|cancelled 🔪 
🖥 desktop 🖥|cancelled 🔪
🍎 iOS 🍎|cancelled 🔪 
🕸 web 🕸|cancelled 🔪

mountiny commented on src/components/HeaderWithBackButton/index.js:
```diff
@@ -0,0 +1,171 @@
+                        textStyles={titleColor ? [StyleUtils.getTextColorStyle(titleColor)] : []}
+                    />
+                )}
+                <View style={[styles.reportOptions, styles.flexRow, styles.pr5, styles.alignItemsCenter]}>
+                    {shouldShowDownloadButton && (
```
This refactor has caused a deploy blocker https://github.com/Expensify/App/pull/23129 as we have forgot to pass the children which was a change made in the deleted file but probably did not cause morege conflicts or it was missed when resolved.

roryabraham replied:
Yes, my bad I missed that change when resolving conflicts

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/5602375137) to production by https://github.com/jasperhuangg in version: 1.3.42-26 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅

roryabraham commented on src/components/Pressable/PressableWithDelayToggle.js:
```diff
@@ -1,22 +1,19 @@
-
-    ...withDelayToggleButtonStatePropTypes,
+    /** Ref passed to the component by React.forwardRef (do not pass from parent) */
+    innerRef: PropTypes.oneOfType([PropTypes.func, PropTypes.shape({current: PropTypes.instanceOf(React.Component)})]).isRequired,
```
My reasoning was that since it's always passed by `forwardRef`, then it would make sense to have it be required. However I see now that it will be null/undefined if the parent doesn't pass a ref.

sobitneupane commented:
[This minor issue](https://github.com/Expensify/App/issues/26425) was probably missed in the PR.

Issue: Preferences - Shows blue color in Bottom Safearea when going offline

Steps to reproduce:

1. Go offline in IOS/native
2. Go to settings -> Go to preferences
3. See bottom Safearea
